Source/SVWidgetsLib/Widgets/ImportASCIIDataWizard/DataFormatPage.cpp

```diff
@@ -698,7 +705,12 @@ void DataFormatPage::on_startRowSpin_valueChanged(int value)
   bool spaceAsDelimiter = field("spaceAsDelimiter").toBool();
   bool consecutiveDelimiters = field("consecutiveDelimiters").toBool();
 
-  QStringList lines = ImportASCIIDataWizard::ReadLines(m_InputFilePath, value, ImportASCIIDataWizard::TotalPreviewLines);
+  // Get our lines from the file cache.
+  QStringList lines;
+  for(int i = value-1; i < m_PreviewLinesCache.size(); i++)
+  {
+    lines.push_back(m_PreviewLinesCache[i]);
+  }
```

Comment:
Is there any way value can be 0?  This method does not have any guards against it, and QStringList does not define what behavior is for [-1], only that "_i_ must be a valid index position in the list".

Reply:
The value passed in _should_ be coming from a QSpinBox that has a minimum set to 1. So in theory value should always be at least 1. But things change and someone else might call that method with a zero.